internal/datastore/common/sql.go

```diff
@@ -275,6 +275,32 @@ func (sqf SchemaQueryFilterer) MustFilterToResourceIDs(resourceIds []string) Sch
 	return updated
 }
 
+// FilterWithResourceIDPrefix returns new SchemaQueryFilterer that is limited to resources whose ID
+// starts with the specified prefix.
+func (sqf SchemaQueryFilterer) FilterWithResourceIDPrefix(prefix string) (SchemaQueryFilterer, error) {
+	if strings.Contains(prefix, "%") {
+		return sqf, spiceerrors.MustBugf("prefix cannot contain the percent sign")
+	}
+	if prefix == "" {
+		return sqf, spiceerrors.MustBugf("prefix cannot be empty")
+	}
+
+	sqf.queryBuilder = sqf.queryBuilder.Where(sq.Like{sqf.schema.colObjectID: prefix + "%"})
```

Comment:
Do we understand if this is properly covered by an index?

When cursors were introduced, having the various datastores match a reasonable index with `ReadRelationships` calls became rather unpredictable. I suspect adding another condition could further confuse query planners. It's worth testing various query scenarios for index coverage. We have some tests that support asserting on EXPLAIN statements (only postgres, unfortunately)

Reply:
The wildcard will be because it is a prefix, but see the note at the top of the PR about indexes overall